packages/plugin-legacy/package.json

```diff
@@ -17,7 +17,6 @@
   "types": "./dist/index.d.ts",
   "exports": {
     ".": {
-      "types": "./dist/index.d.ts",
```

Comment:
What is the reasoning behind removing this entry? We have the same setup on all other plugins: https://github.com/vitejs/vite-plugin-vue/blob/93c444cb99d64c65b71050d0f6a5e3016f7046a1/packages/plugin-vue/package.json#L14

Reply:
TypeScript is able to pick up the adjacent `.ts` by removing this, .e.g `./dist/index.mjs` -> `./dist/index.d.mts`. What this PR does is that it emits two dts files for the two formats :(

Reply:
https://github.com/unjs/unbuild/pull/273

Because of .d.mts and .d.cts, removing it will allow TS to find correct one by itself

Reply:
According to https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-7.html#packagejson-exports-imports-and-self-referencing , the only way to correct provide types for both Node ESM and CJS is to have two separate declaration files, so we need to generate index.d.cts and index.d.mts.

unbuild@2 could generate index.d.cts and index.d.mts standing for types of index.cjs and index.mjs, and could generate index.d.ts for legacy typescript usage.

releated commit:
https://github.com/vuejs/core/commit/d621d4c646b2d7b190fbd44ad1fd04512b3de300

releated PRs:
https://github.com/unjs/unbuild/pull/273
https://github.com/vitejs/vite-plugin-vue/pull/179

Reply:
I think this PR is good for plugins given our current setup. We could release it in a minor for the plugin. I added it to the 5.0 milestone (that we're going to start working on soon) because there is a parallel discussion about moving vite core and the plugins to ESM only. I still don't know if Vite 5 is the right major to do this, so if we don't move forward with that, we should merge this PR instead.
Polls that @bluwy started for reference:
- https://twitter.com/bluwyoo/status/1681341833581056001
- https://elk.zone/m.webtoo.ls/@bluwy/110736050873446169